Realm/ObjectServerTests/RLMAsymmetricSyncServerTests.mm

```diff
@@ -0,0 +1,163 @@
+// the schema. We are only running this tests on the ObjectServer target.
+#if REALM_SERVER_TEST_TARGET
+
+#pragma mark PersonAsymmetric
+
+@interface PersonAsymmetric : RLMAsymmetricObject
+@property RLMObjectId *_id;
+@property NSString *firstName;
+@property NSString *lastName;
+@property NSInteger age;
+
+- (instancetype)initWithPrimaryKey:(RLMObjectId *)primaryKey firstName:(NSString *)firstName lastName:(NSString *)lastName age:(NSInteger)age;
+@end
+
+@implementation PersonAsymmetric
+
++ (NSDictionary *)defaultPropertyValues {
+    return @{@"_id": [RLMObjectId objectId]};
+}
+
++ (NSString *)primaryKey {
+    return @"_id";
+}
+
++ (NSArray *)requiredProperties {
+    return @[@"firstName", @"lastName", @"age"];
+}
+
+- (instancetype)initWithPrimaryKey:(RLMObjectId *)primaryKey firstName:(NSString *)firstName lastName:(NSString *)lastName age:(NSInteger)age  {
+    self = [super init];
+    if (self) {
+        self._id = primaryKey;
+        self.age = age;
+        self.firstName = firstName;
+        self.lastName = lastName;
+    }
+    return self;
+}
+@end
+
+@interface RLMAsymmetricSyncServerTests : RLMSyncTestCase
+@end
+
+@implementation RLMAsymmetricSyncServerTests
+- (void)tearDown {
+    RLMUser *user = [self logInUserForCredentials:[RLMCredentials anonymousCredentials]
+                                              app:self.flexibleSyncApp];
+    RLMMongoClient *client = [user mongoClientWithServiceName:@"mongodb1"];
+    RLMMongoDatabase *database = [client databaseWithName:@"test_data"];
+    RLMMongoCollection *collection = [database collectionWithName:@"PersonAsymmetric"];
+    [self cleanupRemoteDocuments:collection];
+    [super tearDown];
+}
+
+- (void)checkCountInMongo:(unsigned long)count {
+    RLMUser *user = [self logInUserForCredentials:[RLMCredentials anonymousCredentials]
+                                              app:self.flexibleSyncApp];
+    RLMMongoClient *client = [user mongoClientWithServiceName:@"mongodb1"];
+    RLMMongoDatabase *database = [client databaseWithName:@"test_data"];
+    RLMMongoCollection *collection = [database collectionWithName:@"PersonAsymmetric"];
+    XCTestExpectation *findExpectation4 = [self expectationWithDescription:@"should not find documents"];
+    [collection findWhere:@{}
+               completion:^(NSArray<NSDictionary *> *documents, NSError *error) {
+        XCTAssertEqual(documents.count, count);
+        XCTAssertNil(error);
+        [findExpectation4 fulfill];
+    }];
+    [self waitForExpectationsWithTimeout:60.0 handler:nil];
+}
+
+- (void)testOpenLocalRealmWithAsymmetricObjectError {
+    RLMRealmConfiguration *configuration = [RLMRealmConfiguration defaultConfiguration];
+    NSError *error;
+    RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration error:&error];
+    XCTAssertNil(realm);
+    XCTAssertNotNil(error);
+}
+
+- (void)testCreateAsymmetricObjectWithPBSConfigurationShouldNotSync {
```

Comment:
Shouldn't this be an error instead of just silently not working?

Reply:
After discussing this with core, the validation is gonna be done on their side.  I'm going to comment both tests and add a FIXME so when the core release is ready we can enable this tests.